slider/beatmap.py

```diff
@@ -420,7 +444,25 @@ def parse(cls, data, timing_points, slider_multiplier, slider_tick_rate):
         else:
             raise ValueError(f'unknown type code {type_!r}')
 
-        return parse(Position(x, y), time, hitsound, rest)
+        # new combo info is in second bit (0-indexed)
+        new_combo = bool(type_ & (1 << 2))
+        # abusing the int constructor which takes a string and a base. blame
+        # python for not having sane bitarray handling.
+        bitarray = []
```

Comment:
I'm a bit shocked by this implementation of getting the combo skip. My implementation already calculated it in one line...

Reply:
if you know that integers are basically bit arrays, you can just do `type_ & 112` to select bits 4, 5, and 6, and then do `>> 4` to shift the 3 bits to the right, giving you the combo skip value.

Reply:
the bitarray stuff is just stupid, don't get me wrong. would be very happy to get rid of that. I was mostly trying to make the code more explicit - it's pretty hard to tell where the magic number `112` comes from in `type_ & 112`.

I changed it to your version, but with an explicit mask - does that read better?

I don't often work with bit operations, so things which are likely idiomatic and obvious (shifting to get the desired portion of the integer) seem confusing to me at first and worth replacing (even if they aren't). I agree with your code more now that I've had a chance to brush up on my bitwise stuff.